### PR TITLE
[MX-297] - Add city guide entry in search behind flag

### PR DESCRIPTION
**Pull request by brainbicycle**
https://artsyproduct.atlassian.net/browse/MX-297
https://artsyproduct.atlassian.net/browse/MX-298

- Add city guide entry in search behind flag
- Add city guide background image
- Adjustments to city guide UI to accommodate back button

@samchieng 

**Search Entry (Updated Background):**

cityGuidUpdateBG

**Modal Updates**:

cityGuideUpdatedModal

**City Selector Button Updates**:

cityGuideUpdatedButtons

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Scenes/Search/__tests__/SearchEmptyState-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Scenes/Search/Search.tsx [hunk 1]
```diff
@@ -104,6 +113,13 @@ const EmptyState: React.FC<{}> = ({}) => {
   )
 }
 
+const EmptyState: React.FC<{}> = ({}) => {
+  const moveCityGuideEnableSales = NativeModules.Emission.options.AROptionsMoveCityGuideEnableSales
+  const platformIOS = Platform as PlatformIOSStatic
+  const shouldShowCityGuide = moveCityGuideEnableSales && !platformIOS.isPad
```
Comment (ashfurrow) [inline src/lib/Scenes/Search/Search.tsx:119]: Nice 👍

File: src/lib/Scenes/Search/__tests__/Search-tests.tsx [hunk 2]
```diff
@@ -35,6 +37,30 @@ describe("The Search page", () => {
...
+    Object.defineProperty(Platform, "isPad", {
+      get: jest.fn(() => false),
+    })
+    const tree = ReactTestRenderer.create(<TestWrapper />)
+    expect(extractText(tree.root.findByType(SearchEmptyState))).toContain("Explore Art on View by City")
+  })
+
+  it(`does not show city guide entrance when on iPad`, async () => {
+    NativeModules.Emission.options.AROptionsMoveCityGuideEnableSales = true
+    Object.defineProperty(Platform, "isPad", {
```
Comment (ashfurrow) [inline src/lib/Scenes/Search/__tests__/Search-tests.tsx:51]: Oh neat, that makes sense. I didn't realize you could do this, actually!

I think that the `Platform` import in the `Search` component might be better using `isPad` from [`hardware.ts`](https://github.com/artsy/eigen/blob/master/src/lib/Scenes/Artwork/hardware.ts). Then we can use a more standard Jest mock at the top of this file:

```ts
jest.mock("../../hardware", () => ({
  isPad: jest.fn(),
}))
import { isPad } from "../../hardware"

// Later on in a test...

;(isPad as jest.Mock).mockImplementationOnce(() => true)
```

Or something like that. Let me know if you want to pair 👍

Comment (ashfurrow) [reply src/lib/Scenes/Search/__tests__/Search-tests.tsx:51]: We should probably also move `hardware.ts` to a higher-level directory so it's easier to find 😅

**Comment by samchieng** [comment]
@brainbicycle This matches the design but i think i made the image too short in relation to the card. Is it ok if i cut an other image for this tomorrow? Also would be curious what this looks like on an iPad

**Comment by brainbicycle** [comment]
We currently don’t support city guide on iPad so this will only be shown for iPhone, think there would be a fair amount of work involved in getting city guide to work on iPad but I haven’t looked into too much

**Comment by samchieng** [comment]
Ah, that's actually a relief! Good to know.

File: src/lib/Scenes/Search/__tests__/Search-tests.tsx (hunk 2 above)
Comment (brainbicycle) [reply src/lib/Scenes/Search/__tests__/Search-tests.tsx]: 👍

File: Artsy/View_Controllers/Util/ARNavigationController.m [hunk 3]
```diff
@@ -245,7 +245,7 @@ - (void)updateBackButtonPositionForViewController:(UIViewController *)viewContro
         topMargin = self.view.safeAreaInsets.top + 12;
     }
 
-    BOOL fullBleed = [viewController isKindOfClass:ARComponentViewController.class] && [(ARComponentViewController *)viewController fullBleed];
+    BOOL fullBleed = [viewController.class conformsToProtocol:@protocol(ARComponentFullBleedViewSizing)] && [(UIViewController <ARComponentFullBleedViewSizing> *)viewController fullBleed];
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Util/ARNavigationController.m:248]: Nice 👍

**Comment by brainbicycle** [comment]
> Looks good! Thanks for cleaning up the `hardware` file 👍
> 
> Have we tested on a 4" screen simulator? Merge when you're ready 🙌

Good call, looks okay!

iPhoneSECityGuide

iPhoneSECityGuideModal

**Pull request merged**